src/route/dataset-route.ts

```diff
@@ -0,0 +1,176 @@
+import { Request, Response, Router } from 'express';
+import multer from 'multer';
+import pino from 'pino';
+
+// eslint-disable-next-line import/no-cycle
+import { processCSV, uploadCSV, DEFAULT_PAGE_SIZE } from '../controllers/csv-processor';
+import { DataLakeService } from '../controllers/datalake';
+import { Dataset } from '../entity/dataset';
+import { Datafile } from '../entity/datafile';
+import { FileDescription } from '../models/filelist';
+import { datasetToDatasetDTO } from '../dtos/dataset-dto';
+
+const storage = multer.memoryStorage();
+const upload = multer({ storage });
+
+export const logger = pino({
+    name: 'StatsWales-Alpha-App',
+    level: 'debug'
+});
+
+export const apiRoute = Router();
+
+apiRoute.post('/', upload.single('csv'), async (req: Request, res: Response) => {
+    if (!req.file) {
+        res.status(400);
+        res.json({
+            success: false,
+            headers: undefined,
+            data: undefined,
+            errors: [
+                {
+                    field: 'csv',
+                    message: 'No CSV data available'
+                }
+            ]
+        });
+        return;
+    }
+    const internalName: string = req.body?.internal_name;
+    if (!internalName) {
+        res.status(400);
+        res.json({
+            success: false,
+            errors: [
+                {
+                    field: 'internal_name',
+                    message: 'No internal name for the dataset has been provided'
+                }
+            ]
+        });
+        return;
+    }
+    const dataset = Dataset.createDataset(internalName, 'BetaUser');
```

Comment:
So as a route, it should only be used to create a dataset.  I probably need to move the logic out of the route in to either a controller or reusable function somewhere.  I'd say this level of refactoring will happen as we add more end points to better support the user journey.

I agree it will require permissions... When we have Authentication and a clearer idea of RBAC I shall revisit this.